vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/SocketConnectionBase.java

```diff
@@ -161,6 +161,14 @@ private void checkPending() {
           paused = true;
           inflight++;
           cmd = prepareCmd;
+        } else {
+          // cached prepared statement needs to prepare the parameters before execution
+          String msg = queryCmd.prepare();
+          if (msg != null) {
+            inflight--;
+            queryCmd.fail(new NoStackTraceThrowable(msg));
+            return;
```

Comment:
why is return needed here ?

Reply:
It seems this patch of connection base seems more general than this specific issue